<?xml version="1.0" encoding="utf-8"?>
<!-- $Revision: $ -->
<!-- EN-Revision: 52bf027d4cb01fee7d4e33095d3c5ecd6f76fff1 Maintainer: sergey Status: ready -->
<!-- Reviewed: no -->

<refentry xml:id="parallel-future.value" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
 <refnamediv>
  <refname>parallel\Future::value</refname>
  <refpurpose>Разрешение</refpurpose>
 </refnamediv>

 <refsect1 role="description" audience="resolution">
  &reftitle.description;
  <methodsynopsis>
   <modifier>public</modifier> <type>mixed</type><methodname>parallel\Future::value</methodname>
   <void />
  </methodsynopsis>
  <para>
   Возвращает (и при необходимости дожидается) возврата из задания.
  </para>
 </refsect1>

 <refsect1 role="exceptions">
  <title>Исключения</title>
  <warning>
   <para>
    Выбрасывает <type>parallel\Future\Error</type>, если ожидание не удалось (внутренняя ошибка).
   </para>
  </warning>
  <warning>
   <para>
    Выбрасывает <type>parallel\Future\Error\Killed</type>, если <type>parallel\Runtime</type> выполнение задачи было прервано.
   </para>
  </warning>
  <warning>
   <para>
    Выбрасывает <type>parallel\Future\Error\Cancelled</type>, если задача была отменена.
   </para>
  </warning>
  <warning>
   <para>
    Выбрасывает <type>parallel\Future\Error\Foreign</type>, если задача вызвала нераспознанное неперехваченное исключение.
   </para>
  </warning>
  <warning>
   <para>
    Выбрасывает <type>Throwable</type>, не пойманные в задаче.
   </para>
  </warning>
 </refsect1>


</refentry>
<!-- Keep this comment at the end of the file
Local variables:
mode: sgml
sgml-omittag:t
sgml-shorttag:t
sgml-minimize-attributes:nil
sgml-always-quote-attributes:t
sgml-indent-step:1
sgml-indent-data:t
indent-tabs-mode:nil
sgml-parent-document:nil
sgml-default-dtd-file:"~/.phpdoc/manual.ced"
sgml-exposed-tags:nil
sgml-local-catalogs:nil
sgml-local-ecat-files:nil
End:
vim600: syn=xml fen fdm=syntax fdl=2 si
vim: et tw=78 syn=sgml
vi: ts=1 sw=1
-->
